Remove @override annotation from templates and samples

mit-mit opened the pull request:
This removes the `@override` annotation from the Flutter templates and samples. This simplifies the code new users read, and if desired they can add this themselves once their code base grows.

mit-mit committed:
Remove @override from templates

mit-mit committed:
Remove @override in examples

mit-mit commented:
WDYT @Hixie ?

Hixie commented:
I don't think we should change our examples, having different analysis options in different parts of the same repo is confusing.

The change to the template LGTM so long as we work with Tao to have that tested in the near future.

mit-mit commented:
Examples are critical customer facing code, so I'm definitely of the strong opinion that we should optimize for their needs, not our own. And as discussed in person, `@override` is an unneeded complexity and distraction for most developers, especially so for new ones.

I'd suggest we remove `@override` everywhere if you are worried about differentiating.

Hixie commented:
The problem is that `@override` isn't unneeded complexity. It's how we make sure that we keep our examples up to date and correct. Without it, if we change an underlying API, we would have no indication that we should update our examples. To be honest, while I understand some people think it's a "distraction for most developers", I would strongly urge anyone developing for Flutter to use `@override` and have the relevant lint enabled, because they will have more bugs if they don't, and those bugs will be very hard to track down.

IF we commit to usability-testing the template change, and if the change is positive, then I can live with us removing it there (though I wouldn't recommend it). However, for the examples we rely on the annotation, and we can't remove it. We would be risking harm to our developers (because our examples would get out of date).

Removing `@override` everywhere is wildly out of the realm of reasonable. It would dramatically worsen the quality of our APIs and introduce bugs all over the place very quickly.

mit-mit committed:
Revert "Remove @override in examples"

This reverts commit d7d8ca5b0bf1d62486ea56301379c3cbbff1cbdc.

mit-mit commented:
Reverted the change to /examples/.

I'm not sure what you have in mind wrt. testing the template change?

Would you test that developers still understand that the build/overrideState methods override base methods? That's how inheritance works in Dart, so that seems like an odd thing to test.

From my PoV this is a clear usability improvement. If anything, we should have tested the opposite -- when we added `@override` we should have tested that developers understood what that annotation meant. I don't believe we ever did that?

mit-mit commented:
Closing; `@override` will be replaced with `override` as part of null safety.
